### PR TITLE
Remove code-duplication from NativeLibraryLoader

### PR DESCRIPTION
**Pull request by normanmaurer**
Motivation:

NativeLibraryLoader has some code-duplication that can be removed.

Modifications:

Remove duplicated code and just use provided methods of PlatformDependent.

Result:

Less code duplication, fixes [#3756].

### CODE REVIEW

File: common/src/main/java/io/netty/util/internal/PlatformDependent.java [hunk 1]
```diff
@@ -1075,7 +1075,7 @@ private static File tmpdir0() {
     }
 
     @SuppressWarnings("ResultOfMethodCallIgnored")
-    private static File toDirectory(String path) {
+    static File toDirectory(String path) {
```
Comment (Scottmitch) [inline common/src/main/java/io/netty/util/internal/PlatformDependent.java]: is this change necessary?

File: common/src/main/java/io/netty/util/internal/NativeLibraryLoader.java [hunk 2]
```diff
@@ -43,13 +43,13 @@
...
-    private static final String OSNAME;
+    private static final boolean OSX;
     private static final File WORKDIR;
     private static final boolean DELETE_NATIVE_LIB_AFTER_LOADING;
 
     static {
-        OSNAME = SystemPropertyUtil.get("os.name", "").toLowerCase(Locale.US).replaceAll("[^a-z0-9]+", "");
-
+        String osname = SystemPropertyUtil.get("os.name", "").toLowerCase(Locale.US);
+        OSX = osname.startsWith("macosx") || osname.startsWith("osx");
```
Comment (Scottmitch) [inline common/src/main/java/io/netty/util/internal/NativeLibraryLoader.java]: would it make sense to move the OS X platform detection to `PlatformDependent`?

File: common/src/main/java/io/netty/util/internal/PlatformDependent.java (hunk 1 above)
Comment (normanmaurer) [inline common/src/main/java/io/netty/util/internal/PlatformDependent.java]: yes as we use it in ´NativeLibraryLoader´

File: common/src/main/java/io/netty/util/internal/NativeLibraryLoader.java (hunk 2 above)
Comment (normanmaurer) [inline common/src/main/java/io/netty/util/internal/NativeLibraryLoader.java]: sure we can do but as there were no other places where this is used I did not bother.

File: common/src/main/java/io/netty/util/internal/PlatformDependent.java (hunk 1 above)
Comment (Scottmitch) [inline common/src/main/java/io/netty/util/internal/PlatformDependent.java]: It looks like you removed all usages in this PR, or am I missing something?

File: common/src/main/java/io/netty/util/internal/NativeLibraryLoader.java (hunk 2 above)
Comment (Scottmitch) [inline common/src/main/java/io/netty/util/internal/NativeLibraryLoader.java]: Yip understood. My rational is if we need todo macOS specific stuff in the future `PlatformDependent` is the natural place to look, but we may miss it (and duplicate it) if the logic lives in this class. Since `PlatformDependent` is internal and duplication removal was the motivation for this PR it seemed like it warranted a discussion.

File: common/src/main/java/io/netty/util/internal/PlatformDependent.java (hunk 1 above)
Comment (normanmaurer) [inline common/src/main/java/io/netty/util/internal/PlatformDependent.java]: if that is true... which may be true that at the end I was able to remove all of them you are right... let me check

File: common/src/main/java/io/netty/util/internal/NativeLibraryLoader.java (hunk 2 above)
Comment (normanmaurer) [inline common/src/main/java/io/netty/util/internal/NativeLibraryLoader.java]: true, let me move it.

File: common/src/main/java/io/netty/util/internal/PlatformDependent.java (hunk 1 above)
Comment (normanmaurer) [inline common/src/main/java/io/netty/util/internal/PlatformDependent.java]: ok you are right... thanks :)

**Comment by normanmaurer** [comment]
Cherry-picked into 4.1 (bdb0a39c8a) and 4.0 (1a0ef664c5d5371654c5a1680fb11df0ddd5efeb)
